### ISSUE TITLE
[Bug]: react/prop-types error detected on linux but won't on windows

### ISSUE THREAD

**Issue by fauzanashari** [jsx-eslint/eslint-plugin-react#3702]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

windows detect `react/prop-types` error for only client component, and linux detect error from server and client component

windows:
![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/114401788/623bb061-5f3d-4b71-bd8a-151824870ab8)

linux:
![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/114401788/20c10418-93e7-4cf6-8150-829c73b654a7)

I am develop nextjs 13 app with server component & client component without defining propTypes for both, but `npm run lint` only the detect error for client component. I cloned same code base on Fedora 38 and unlike window `npm run lint` detect error for both client and server component

`npm run lint`
`npx next build`

### Expected Behavior

It should be output same error both windows or linux, since many developers use windows as dev env and linux as production.

windows:
![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/114401788/623bb061-5f3d-4b71-bd8a-151824870ab8)

linux:
![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/114401788/20c10418-93e7-4cf6-8150-829c73b654a7)

### eslint-plugin-react version

v7.33.2

### eslint version

v8

### node version

v18.19.1

**Comment by ljharb** [comment]
Can you confirm on the command line that this behavior occurs (vscode isn’t the source of truth), and that both envs have the same deps at the same versions installed on disk?

**Comment by fauzanashari** [comment]
> Can you confirm on the command line that this behavior occurs (vscode isn’t the source of truth), and that both envs have the same deps at the same versions installed on disk?

linux
![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/114401788/519c9e4c-a3ba-4b9d-b829-36ccecfd9972)

windows
![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/114401788/b872d593-44e0-499e-882a-b88c97e89a12)

both files location on `./src/app/role-permission/ubah/[id]/page.jsx` and you can see in windows it doesn't even throw an error from these file

**Comment by erlanggaht** [comment]
The bug is in all Linux versions, mas ?

**Comment by ljharb** [comment]
@fauzanashari what is `[id]`? is it literal square brackets (which might break many file system tools), or is it a placeholder for something else?

**Comment by fauzanashari** [comment]
> @fauzanashari what is `[id]`? is it literal square brackets (which might break many file system tools), or is it a placeholder for something else?

yes nextjs this is nextjs convention to create dynamic slug url https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes

**Comment by ljharb** [comment]
interesting - i wonder if that's the issue. I'll try to create a test case to rule that out.
